### PR TITLE
Dev/scheme selection on each column

### PR DESCRIPTION
**Pull request by teisvh**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1,17 +1,57 @@
...
+#include "models/Gorilla.h"
+#include "doctest.h"
+#include "managers/ConfigManager.h"
+#include "managers/ReaderManager.h"
+
+// This function is required by Doctest to make it possible to have asserts inside the code.
+// The function is passed to the Doctest context in our main function.
+static void handler(const doctest::AssertData& ad) {
+    using namespace doctest;
+
```
Comment (karljohan12) [inline src/main.cpp:15]: Maybe source

File: src/managers/TimestampManager.cpp [hunk 2]
```diff
@@ -266,63 +301,238 @@ std::vector<uint8_t> TimestampManager::binaryCompressLocOffsets(
...
+}
+
+
+std::vector<int>
+TimestampManager::decompressLocalOffsetList(std::vector<uint8_t> values) {
+
+    std::map<int, std::vector<std::tuple<int, int>>> offsetsDecompressed;
+    std::vector<int> temp;
+    std::vector<int> decompressed;
+    BitReader bitReader = tryNewBitReader(values, values.size());
```
Comment (karljohan12) [inline src/managers/TimestampManager.cpp]: Why try?

File: src/managers/TimestampManager.cpp [hunk 3]
```diff
@@ -266,63 +301,238 @@ std::vector<uint8_t> TimestampManager::binaryCompressLocOffsets(
...
+    int schemeID = readBits(&bitReader, 8);
+
+    for (int i = 0; i < lolSize - 1; i++) {
+
+        // decompressNextValue returns 'false' if no control codes are found, i.e. the zero bit at
+        // the end of each column.
+        // When that happens, read the next eight bits to receive new schemeID
+        if (!decompressNextValue(schemes.at(schemeID), &bitReader, &currentValue, &decompressed)) {
+            int schID = readBits(&bitReader, 8);
+            schemeID = schID;
```
Comment (karljohan12) [inline src/managers/TimestampManager.cpp]: schemeID = readBits(&bitReader, 8);

**Review by karljohan12 — CHANGES_REQUESTED** [review]

File: src/main.cpp (hunk 1 above)
Comment (teisvh) [reply src/main.cpp:15]: fixed

File: src/managers/TimestampManager.cpp (hunk 2 above)
Comment (teisvh) [reply src/managers/TimestampManager.cpp]: refactored to a constructor

File: src/managers/TimestampManager.cpp (hunk 3 above)
Comment (teisvh) [reply src/managers/TimestampManager.cpp]: fixed

File: src/managers/TimestampManager.cpp (hunk 2 above)
Comment (karljohan12) [reply src/managers/TimestampManager.cpp]: <3

**Review by karljohan12 — APPROVED** [review]

File: src/managers/TimestampManager.cpp [hunk 4]
```diff
@@ -266,63 +301,237 @@ std::vector<uint8_t> TimestampManager::binaryCompressLocOffsets(
...
+    int currentSchemeVal = 0;
+
+    // Read the control bits. The last control code of all schemes have the same length as the
+    // second to last code, but the last bit is '1' instead of '0'.
+    // Condition #1 ensures that we correctly read the last control code correctly.
+    //      This is the first condition, as it short-circuits if false (which is intentional)
+    // Condition #2 reads the bits and breaks out of the loop when a 0-bit is encountered in
+    //      the control bit. The 0-bit is also consumed.
+    int i = 0;
+    for (; i < schemeVals.size() && readBit(bitReader); i++) {}
```
Comment (agneborn98) [inline src/managers/TimestampManager.cpp]: Er det her et tomt loop???

File: src/managers/TimestampManager.cpp [hunk 5]
```diff
@@ -266,63 +301,237 @@ std::vector<uint8_t> TimestampManager::binaryCompressLocOffsets(
...
             }
         }
 
+        globID++;
+    }
+
+    // The following code finds the best compression scheme for compressing the size of the local
+    // offset list
+
+    int lolSize = getSizeOfLocalOffsetList();
```
Comment (agneborn98) [inline src/managers/TimestampManager.cpp:363]: Hvorfor ikke bare .size() i stedet for dette? Er det ikke en container?

**Review by agneborn98 — CHANGES_REQUESTED** [review]

**Review by agneborn98 — APPROVED** [review]

**Pull request merged**